### ISSUE TITLE
jsx-indent: ignore childs under thead, tbody and tfoot

### ISSUE THREAD

**Issue by balthild** [yannickcr/eslint-plugin-react#2266]
```html
<table>
    <thead>
    <tr>
        <th>Name</th>
        <th>Description</th>
    </tr>
    </thead>
    <tbody>
    <tr>
        <td>xxxx</td>
        <td>yyyy</td>
    </tr>
    </tbody>
</table>
```

`<thead>` and `<tr>` has the same indentation level, which is complained by the linter:

```
  103:25  error  Expected indentation of 28 space characters but found 24  react/jsx-indent
  112:25  error  Expected indentation of 28 space characters but found 24  react/jsx-indent
  119:25  error  Expected indentation of 28 space characters but found 24  react/jsx-indent
  126:25  error  Expected indentation of 28 space characters but found 24  react/jsx-indent
```

We can add an option `noIndentChild` accepts a list of element names. For example:

```js
{
  "react/jsx-indent": ["error", 4, { 
    noIndentChild: ["thead", "tbody", "tfoot"] 
  }]
}
```

**Comment by ljharb** [comment]
It seems strange to arbitrarily alter the indentation of certain elements - why would you not indent all of those equally?

**Comment by balthild** [comment]
It's just a kind of convention, which is also the default indentation settings in IntelliJ IDEA.

**Comment by ljharb** [comment]
I don’t think it would make sense as a generic option due to the complexity involved; and it seems a bit overkill to add an option specifically for these three table tags, especially when it’s a very uncommon convention (your editor shouldn’t have any opinion about formatting; that’s what the linter and autofix is for).

**Comment by balthild** [comment]
Since IntelliJ IDEA (or WebStorm) is one of the most popular IDEs, it's default formatting preference might not be something really uncommon... But that's alright, I could change the settings to fit eslint for the tie being.
